app/legacy/v2/notifications/route_schema.py

```diff
@@ -103,22 +112,54 @@ class V2GetSmsNotificationResponseModel(V2GetNotificationResponseModel):
 ##################################################
 
 
+class PersonalisationFileObject(BaseModel):
+    """Personalisation file attachment object."""
+
+    file: str
+    filename: str = Field(..., min_length=3, max_length=255)
+    sending_method: AttachmentSendingMethodType | None = None
+
+
 class V2PostNotificationRequestModel(BaseModel):
     """Common attributes for the POST /v2/notifications/<:notification_type> routes request."""
 
     billing_code: str | None = Field(max_length=256, default=None)
     callback_url: HttpUrl | None = Field(max_length=255, default=None)
-    personalisation: dict[str, str | int | float] | None = None
+    personalisation: dict[str, Union[str, List[str], PersonalisationFileObject]] | None = None
```

Comment:
Was there a problem using the new-style union? Using `Union` isn't usually necessary in newer versions of python.
```suggestion
    personalisation: dict[str, str | List[str] | PersonalisationFileObject] | None = None
```

Reply:
You're right, new-style works.